From e2639cc89101ba1e5bb29dffda7295c2dd3bc11e Mon Sep 17 00:00:00 2001
From: Giulio Cervera <giulio.cervera@gmail.com>
Date: Mon, 9 Jan 2012 19:48:33 +0100
Subject: [PATCH] msm: kgsl: fix after lastest merge

* fix typo
* we still use older file/var name
* cleanup Makefile
* remove unused var
---
 drivers/gpu/msm/Makefile            |    6 ------
 drivers/gpu/msm/adreno_ringbuffer.c |    6 +++---
 drivers/gpu/msm/kgsl.c              |    3 +--
 3 files changed, 4 insertions(+), 11 deletions(-)

diff --git a/drivers/gpu/msm/Makefile b/drivers/gpu/msm/Makefile
index 6ae89b9..2fad011 100644
--- a/drivers/gpu/msm/Makefile
+++ b/drivers/gpu/msm/Makefile
@@ -18,14 +18,8 @@ msm_adreno-$(CONFIG_GPU_MSM_KGSL) += \
 
 msm_adreno-$(CONFIG_DEBUG_FS) += adreno_debugfs.o
 
-msm_z180-$(CONFIG_MSM_KGSL_2D) += \
-	kgsl_g12_drawctxt.o \
-	kgsl_g12_cmdstream.o \
-	kgsl_g12.o
-
 msm_kgsl_core-objs = $(msm_kgsl_core-y)
 msm_adreno-objs = $(msm_kgsl_adreno-y)
-msm_z180-objs = $(msm_kgsl_z180-y)
 
 obj-$(CONFIG_GPU_MSM_KGSL) += \
 	msm_kgsl_core.o \
diff --git a/drivers/gpu/msm/adreno_ringbuffer.c b/drivers/gpu/msm/adreno_ringbuffer.c
index 6e1c34f..082425d 100644
--- a/drivers/gpu/msm/adreno_ringbuffer.c
+++ b/drivers/gpu/msm/adreno_ringbuffer.c
@@ -553,8 +553,8 @@ int kgsl_ringbuffer_close(struct kgsl_ringbuffer *rb)
 	kgsl_sharedmem_free(&rb->buffer_desc);
 	kgsl_sharedmem_free(&rb->memptrs_desc);
 
-	kfree(adreno_dev->pfp_fw);
-	kfree(adreno_dev->pm4_fw);
+	kfree(yamato_device->pfp_fw);
+	kfree(yamato_device->pm4_fw);
 
 	yamato_device->pfp_fw = NULL;
 	yamato_device->pm4_fw = NULL;
@@ -675,7 +675,7 @@ int kgsl_ringbuffer_close(struct kgsl_ringbuffer *rb)
 
 	if (device->state & KGSL_STATE_HUNG)
 		return -EBUSY;
-	if (!(yamato_devce->ringbuffer.flags & KGSL_FLAGS_STARTED) ||
+	if (!(yamato_device->ringbuffer.flags & KGSL_FLAGS_STARTED) ||
 	      context == NULL || ibdesc == 0 || numibs == 0)
 		return -EINVAL;
 
diff --git a/drivers/gpu/msm/kgsl.c b/drivers/gpu/msm/kgsl.c
index 6e0be10..7c5d4ea 100644
--- a/drivers/gpu/msm/kgsl.c
+++ b/drivers/gpu/msm/kgsl.c
@@ -243,7 +243,6 @@ static void kgsl_memqueue_freememontimestamp(struct kgsl_device *device,
 static void kgsl_memqueue_drain(struct kgsl_device *device)
 {
 	struct kgsl_mem_entry *entry, *entry_tmp;
-	struct kgsl_event *event, *event_tmp;
 	uint32_t ts_processed;
 
 	BUG_ON(!mutex_is_locked(&device->mutex));
@@ -1767,7 +1766,7 @@ typedef long (*kgsl_ioctl_func_t)(struct kgsl_device_private *,
 	KGSL_IOCTL_FUNC(IOCTL_KGSL_CFF_USER_EVENT,
 			kgsl_ioctl_cff_user_event, 0),
 	KGSL_IOCTL_FUNC(IOCTL_KGSL_TIMESTAMP_EVENT,
-			kgsl_ioctl_timestamp_event, 0),
+			kgsl_ioctl_timestamp_event, 1),
 };
 
 static long kgsl_ioctl(struct file *filep, unsigned int cmd, unsigned long arg)
-- 
1.7.5.4

